lib/src/yaru_progress_indicator.dart

```diff
@@ -0,0 +1,545 @@
+import 'package:flutter/foundation.dart';
+import 'package:flutter/material.dart';
+import 'dart:math' as math;
+
+const double _kMinCircularProgressIndicatorSize = 36.0;
+const double _kDefaultCircularProgressStrokeWidth = 6;
+const double _kDefaultLinearProgressMinHeight = 6;
+const int _kIndeterminateAnimationDuration = 8000;
+const Curve _kIndeterminateAnimationCurve =
+    Cubic(.35, .75, .65, .25); // Kind of `Curves.slowMiddle` curve
+
+abstract class _YaruProgressIndicator extends StatefulWidget {
+  /// Creates a Yaru progress indicator.
+  ///
+  /// {@macro flutter.material.ProgressIndicator.ProgressIndicator}
+  const _YaruProgressIndicator({
+    Key? key,
+    this.value,
+    this.color,
+    this.valueColor,
+    this.semanticsLabel,
+    this.semanticsValue,
+  }) : super(key: key);
+
+  /// If non-null, the value of this progress indicator.
+  ///
+  /// A value of 0.0 means no progress and 1.0 means that progress is complete.
+  /// The value will be clamped to be in the range 0.0-1.0.
+  ///
+  /// If null, this progress indicator is indeterminate, which means the
+  /// indicator displays a predetermined animation that does not indicate how
+  /// much actual progress is being made.
+  final double? value;
+
+  /// {@macro flutter.progress_indicator.ProgressIndicator.color}
+  final Color? color;
+
+  /// The progress indicator's color as an animated value.
+  ///
+  /// If null, the progress indicator is rendered with [color]. If that is null,
+  /// then it will use the ambient [ProgressIndicatorThemeData.color]. If that
+  /// is also null then it defaults to the current theme's [ColorScheme.primary].
+  final Animation<Color?>? valueColor;
+
+  /// {@macro flutter.progress_indicator.ProgressIndicator.semanticsLabel}
+  final String? semanticsLabel;
+
+  /// {@macro flutter.progress_indicator.ProgressIndicator.semanticsValue}
+  final String? semanticsValue;
+
+  Color _getValueColor(BuildContext context) {
+    return valueColor?.value ??
+        color ??
+        ProgressIndicatorTheme.of(context).color ??
+        Theme.of(context).colorScheme.primary;
+  }
+
+  @override
+  void debugFillProperties(DiagnosticPropertiesBuilder properties) {
+    super.debugFillProperties(properties);
+    properties.add(PercentProperty('value', value,
+        showName: false, ifNull: '<indeterminate>'));
+  }
+
+  Widget _buildSemanticsWrapper({
+    required BuildContext context,
+    required Widget child,
+  }) {
+    String? expandedSemanticsValue = semanticsValue;
+    if (value != null) {
+      expandedSemanticsValue ??= '${(value! * 100).round()}%';
+    }
+    return Semantics(
+      label: semanticsLabel,
+      value: expandedSemanticsValue,
+      child: child,
+    );
+  }
+}
+
+class YaruLinearProgressIndicator extends _YaruProgressIndicator {
+  /// Creates a Yaru circular progress indicator.
```

Comment:
Fixed, thanks @mivoligo :)

Reply:
Awesome @Jupi007 :+1: I'm not sure also about those `@macro` in comments. Do we need that or is it from Flutter?

Reply:
Well, as I have the same signature as the material widget, I reused the same comments. And as they define some comment templates I'm able to include them with the `@macro` comment instead of doing a simple copy-paste.

But in hindsight, I think it is a bad idea now, because somebody who just navigate into the code won't be able to see correct comments, and if the template comments are changed, it could not fit our code.
I'll change that so.